app/models/manageiq/providers/embedded_ansible/automation_manager/scm_credential.rb

```diff
@@ -54,6 +54,8 @@ class ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ScmCredential < M
   alias ssh_key_data   auth_key
   alias ssh_key_unlock auth_key_password
 
+  before_save :ensure_newline_for_ssh_key
```

Comment:
would before_validation work as well?

feel like data manipulation should happen before validation rather than before save

Reply:
Seems like it still passes the tests and I have no strong preference myself.

@Fryguy since I expect that you will give a final say anyway, do you want to weigh in on which you prefer before I end up rebasing this multiple times?

Reply:
Hi @NickLaMuro, Would preserving the private key as entered be an option here?  The reason I ask is that an RSA private key without the newline works fine in Ansible.  The problem we encountered was specific to ssh keys generated on 5.11 (rhel 8).  There wasn't a problem when we used ssh keys generated on 5.11 (rhel 8) with the -m PEM option which creates the RSA private key as opposed tp the OPENSSH private key.

Reply:
I think before_validation is the better choice

Reply:
@tinaafitz I tested both types of private keys on a generic `git clone` with the following done to them:

```
$ echo "" >> ~/.ssh/id_rsa
$ echo "" >> ~/.ssh/id_rsa
$ echo "" >> ~/.ssh/id_rsa
```

So any number of newlines applied was no problem.  However, lack of a `\n` seemed to be problematic, so I am airing on the side of "add a new line if it doesn't exist" universally instead of trying to parse specific keys.

If you find a key where this is not the case, we can address, but it seems like the following line:

```
-----END OPENSSH PRIVATE KEY-----
```

And

```
-----END RSA PRIVATE KEY-----
```

Are all that need to be in place, and subsequent lines are ignored.

Reply:
Thanks @NickLaMuro. I feel better about it. :-)